app.py

```diff
@@ -55,24 +55,38 @@ def experience():
     json
         The experience(s) as a JSON object, or an error message.
     '''
+    response = None
     if request.method == 'GET':
         index = request.args.get('index')
         experiences = data.get('experience',[])
         if index is not None:
             try:
                 index = int(index)
                 if 0 <= index < len(experiences):
-                    return jsonify(experiences[index])
-                return jsonify({'error':'Experience does not exist'}), 404
+                    response = jsonify(experiences[index])
+                else:
+                    response = jsonify({'error':'Experience does not exist'}), 404
             except ValueError:
-                return jsonify({'error':'Experience does not exist'}), 404
-        return jsonify(experiences)
+                response = jsonify({'error':'Experience does not exist'}), 404
+        else:
+            response = jsonify(experiences)
 
     if request.method == 'POST':
-        return jsonify({})
-
-    return jsonify({})
-
+        required_fields = ['title','company','start_date','end_date','description','logo']
+        if not request.json:
+            response = jsonify({'error':'Filed Creating Experience'}), 400
```

Comment:
Maybe we should let the user know that they are missing some data instead of generic error "Failed"